### PR TITLE
Issue #1032: Filter UI with dialogs

### LINKED ISSUE #1032: [UX] move text filter configuration into modals
(Related: #975)

https://www.drupal.org/project/drupal/issues/1834682

Before:

screen shot 2018-11-02 at 6 52 37 pm

After:

screen shot 2018-11-02 at 6 54 29 pm

---
~~PR by @docwilmot: https://github.com/backdrop/backdrop/pull/1023~~
~~PR by @docwilmot: https://github.com/backdrop/backdrop/pull/3074~~
~PR by @docwilmot: https://github.com/backdrop/backdrop/pull/3077~
~PR by @docwilmot: https://github.com/backdrop/backdrop/pull/3096 using dialogs~
PR by @docwilmot and @quicksketch: https://github.com/backdrop/backdrop/pull/3131

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/1032

Replaces https://github.com/backdrop/backdrop/pull/3096

---

This builds on @docwilmot's PR by adjusting the way that the hidden "filtered_html" element is updated and the corresponding filter tips on the main form are updated. This should solve issues with editor buttons (such as the image button and strike-through button) automatically updating the allowed tag list.

### CODE REVIEW

**Comment by quicksketch** [comment]
New sandbox after further cleanup.

File: core/modules/filter/css/filter.admin.css [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Styles used when configuring a text format.
+ */
+#filter-order .filter-info {
+  padding: 0 0 0 3em;
```
Comment (jenlampton) [inline core/modules/filter/css/filter.admin.css:5]: We need a RTL version of this.

File: core/modules/filter/filter.module [hunk 2]
```diff
@@ -316,6 +324,40 @@ function filter_format_save($format) {
     $format->filters[$name] = $filter;
   }
 
+  return $format;
+}
+
+/**
+ * Saves a text format object to the database.
```
Comment (jenlampton) [inline core/modules/filter/filter.module]: Aren't these config? Not in the database?

File: core/modules/filter/js/filter.filtered_html.admin.js [hunk 3]
```diff
@@ -56,9 +56,9 @@ Backdrop.behaviors.filterFilterHtmlUpdating = {
 
   attach: function (context, settings) {
     var that = this;
-    $(context).find('[name="filters[filter_html][settings][allowed_html]"]').once('filter-filter_html-updating').each(function () {
+    $(context).find('#allowed-html').once('filter-filter_html-updating').each(function () {
```
Comment (jenlampton) [inline core/modules/filter/js/filter.filtered_html.admin.js:59]: some places use IDs and some use data attributes, is that ok?

Comment (quicksketch) [reply core/modules/filter/js/filter.filtered_html.admin.js:59]: Yeah I think so. I used data attributes where a machine name was being embedded into the DOM element and IDs where we were hard-coding a particular string.

File: core/modules/filter/filter.module (hunk 2 above)
Comment (quicksketch) [reply core/modules/filter/filter.module]: Yeah would be better to say config.

Comment (jenlampton) [reply core/modules/filter/filter.module]: updated.

**Comment by quicksketch** [comment]
Merged into 1.x for 1.16.0. Woot!

**Pull request merged**